test/e2e/framework/types/job.go

```diff
@@ -267,8 +268,11 @@ func (j *Job) validateStep(step *StepWrapper) error {
 		return nil
 
 	default:
+		// validate dns steps
+		if strings.Contains(val.Type().Name(), "DNS") {
```

Comment:
Let's avoid making specific changes like this to the framework types, it's a slippery slope, what's the issue this is solving?

Reply:
Index out of bound when a step contains a struct. The other option is to loop through all structs inside a step and validate them. Is there a contract where steps should not contain structs?

```
	Step: &ValidateDNSRequest{
		Metric: Metric{
			DNSRetinaPort: "10093",
			NumResponse:   "0",
			Query:         "kubernetes.default.svc.cluster.local.",
			QueryType:     "AAAA",
			ReturnCode:    "",
			Response:      "",
		},
	}
```

Reply:
Originally I had the intention that steps would only contain string properties, just as a CLI command would be authored